Tests/Translation/Extractor/PhpExtractorTest.php

```diff
@@ -182,12 +188,112 @@ public function provideFormTranslationFixtures()
         ];
     }
 
-    /**
-     * @param $fixtureResource
-     *
-     * @return MessageCatalogue
-     */
-    private function buildMessageCatalogue($fixtureResource)
+    public function testExtractFromDirectory(): void
+    {
+        $messageCatalogue = $this->buildMessageCatalogue('directory/');
+
+        $catalogue = $messageCatalogue->all();
+        $this->assertCount(2, array_keys($catalogue));
+        $this->assertCount(3, $catalogue['messages']);
+        $this->assertCount(2, $catalogue['admin.product.help']);
+
+        $this->verifyCatalogue($messageCatalogue, [
+            'messages' => [
+                'SecondSubdirShop' => 'SecondSubdirShop',
+                'SubdirShop' => 'SubdirShop',
+                'Shop' => 'Shop',
+            ],
+            'admin.product.help' => [
+                'SubdirFingers' => 'SubdirFingers',
+                'Fingers' => 'Fingers',
+            ],
+        ]);
+
+        $messageCatalogue = new MessageCatalogue('en');
```

Comment:
IMO these are independent tests and should be in separate methods
```suggestion
    }
    
    public function testExtractFromDirectoryWithExclusions(): void
    {
        $messageCatalogue = new MessageCatalogue('en');
```